shared/src/types/storage.rs

```diff
@@ -252,6 +253,23 @@ impl Key {
         }
     }
 
+    /// Check if the given key can be updated
+    pub fn is_updatable(&self) -> bool {
+        match &self.segments[..] {
+            [
+                DbKeySeg::AddressSeg(_),
+                DbKeySeg::StringSeg(key),
+                DbKeySeg::AddressSeg(owner),
+            ] if key == BALANCE_STORAGE_KEY
+                && (*owner == Address::Internal(InternalAddress::Burn)
+                    || *owner == Address::Internal(InternalAddress::Mint)) =>
+            {
+                false
+            }
+            _ => true,
+        }
```

Comment:
slightly more concise
```suggestion
        !matches!(&self.segments[..], [
                DbKeySeg::AddressSeg(_),
                DbKeySeg::StringSeg(key),
                DbKeySeg::AddressSeg(Address::Internal(
                    InternalAddress::Burn | InternalAddress::Mint)),
            ] if key == BALANCE_STORAGE_KEY
        )
```